app/uk/gov/hmrc/individualsdetailsapi/connectors/IfConnector.scala

```diff
@@ -108,26 +99,35 @@ class IfConnector @Inject()(servicesConfig: ServicesConfig, http: HttpClient, va
 
   private def recover[A](x: Future[A], emptyResponse: A, apiIfFailedAuditRequest: ApiIfFailureAuditRequest)
                         (implicit hc: HeaderCarrier, ec: ExecutionContext): Future[A] = x.recoverWith {
+      case validationError: JsValidationException => {
+        Logger.error(s"Error parsing IF response: ${validationError.errors}")
+        auditHelper.auditIfApiFailure(apiIfFailedAuditRequest, s"Error parsing IF response: ${validationError.errors}")
+        Future.failed(new InternalServerException("Something went wrong."))
+      }
       case notFound: NotFoundException => {
-        auditHelper.auditIfApiFailure(apiIfFailedAuditRequest,
-                                      notFound.getMessage)
+        auditHelper.auditIfApiFailure(apiIfFailedAuditRequest, notFound.getMessage)
         Future.successful(emptyResponse)
       }
-      case Upstream4xxResponse(msg, 429, _, _) => {
-        Logger.warn(s"IF Rate limited: $msg")
+      case Upstream5xxResponse(msg, _, _, _) => {
+        Logger.error(s"Upstream internal server error: $msg")
         auditHelper.auditIfApiFailure(apiIfFailedAuditRequest,
-                                      s"IF Rate limited: $msg")
+          s"Internal Server error: $msg")
+        Future.failed(new InternalServerException("Something went wrong."))
+      }
+      case Upstream4xxResponse(msg, 429, _, _) => {
+        Logger.warn(s"IF rate limited: $msg")
+        auditHelper.auditIfApiFailure(apiIfFailedAuditRequest,s"IF Rate limited: $msg")
         Future.failed(new TooManyRequestException(msg))
       }
       case Upstream4xxResponse(msg, _, _, _) => {
+        Logger.error(s"IF returned invalid request: $msg")
```

Comment:
@owen-opencast - I think we need to be carful logging this response to Kibana. When we log upstream 400 errors to Kibana the log entry may also contain too much info for Kibana. For this error and schema errors above Kibana should simply tell us something went wrong with the message detail being written to Splunk only.